File: code/generate_strategies/Direct/java/function_tqdm_desc.java
package com.example;

import me.tongfei.progressbar.ProgressBar;
import me.tongfei.progressbar.ProgressBarBuilder;

public class ProgressBar {
    public String process_with_progress_bar(String data, String desc_text) {
        class TypeError extends RuntimeException {
            public TypeError(String message) {
                super(message);
            }
        }

        class ProgressBarWrapper {
            String process_with_progress_bar(String data, String desc_text) {
                if (desc_text != null && !(desc_text instanceof String)) {
                    throw new TypeError("desc_text must be a string.");
                }

                StringBuilder results = new StringBuilder();
                try (ProgressBar pb = new ProgressBarBuilder()
                        .setTaskName(desc_text != null ? desc_text : "")
                        .setInitialMax(data.length())
                        .build()) {

                    for (char element : data.toCharArray()) {
                        String processed_element = String.valueOf(
                                Integer.parseInt(String.valueOf(element)) * 2
                        );
                        results.append(processed_element);
                        pb.step();
                    }
                }
                return results.toString();
            }
        }

        return new ProgressBarWrapper().process_with_progress_bar(data, desc_text);
    }
}

File: manual/code/generate_strategies/RA(method)/python/function_tqdm_desc.py
from tqdm import tqdm

class ProgressBar:
    @staticmethod
    def process_with_progress_bar(data: str, desc_text: str = None) -> str:
        if desc_text and not isinstance(desc_text, str):
            raise TypeError("desc_text must be a string.")
        
        results = []
        for element in tqdm(data, desc=desc_text):
            processed_element = str(int(element) * 2)
            results.append(processed_element)
        return ''.join(results)
